uint8[] crypt_string
